### PR TITLE
[MRG] Improve notebook tests

### PR DESCRIPTION
**Pull request by fdroessler**
This PR is to further improve the notebook tests by adding the following:

- [x]  Add some notes to CONTRIBUTING.md
- [x]  Include translation notebooks in tests
- [x]  Add tests that checks if all notebooks are either ignored or tested
- [x]  Speed up test suit

### CODE REVIEW

**Comment by fdroessler** [comment]
I'll suggest moving all the translations of the notebooks into a separate folder to avoid a folder explosion in the tutorials root directory.

File: test/notebooks/test_notebooks.py [hunk 1]
```diff
@@ -15,76 +16,140 @@
...
     """Test Notebooks in the tutorial root folder."""
     notebooks = glob.glob("*.ipynb")
     for notebook in notebooks:
-        if Path(notebook).name in exclusion_list_basic:
-            continue
-        res = pm.execute_notebook(
-            notebook, "/dev/null", parameters={"epochs": 1, "n_test_batches": 5}, timeout=300
-        )
-        assert isinstance(res, nbformat.notebooknode.NotebookNode)
+        print(notebook)
```
Comment (LaRiffle) [inline test/notebooks/test_notebooks.py]: extra print here

File: test/notebooks/test_notebooks.py [hunk 2]
```diff
@@ -15,76 +16,140 @@
...
+                timeout=300,
+            )
+            assert isinstance(res, nbformat.notebooknode.NotebookNode)
+
+
+def test_notebooks_basic_translations(isolated_filesystem):
+    """Test Notebooks in the tutorial root folder."""
+    notebooks = glob.glob("translations/**/*.ipynb", recursive=True)
+    for notebook in notebooks:
+        print(notebook)
```
Comment (LaRiffle) [inline test/notebooks/test_notebooks.py]: same

File: test/notebooks/test_notebooks.py [hunk 3]
```diff
@@ -15,76 +16,140 @@
...
 
 def test_notebooks_advanced(isolated_filesystem):
     notebooks = glob.glob("advanced/*.ipynb")
+    notebooks += glob.glob("advanced/Split Neural Network/*.ipynb")
     for notebook in notebooks:
-        if Path(notebook).name in exclusion_list_advanced:
-            continue
-        res = pm.execute_notebook(notebook, "/dev/null", parameters={"epochs": 1}, timeout=300)
-        assert isinstance(res, nbformat.notebooknode.NotebookNode)
+        print(notebook)
```
Comment (LaRiffle) [inline test/notebooks/test_notebooks.py]: same

File: test/notebooks/test_notebooks.py (hunk 1 above)
Comment (LaRiffle) [reply test/notebooks/test_notebooks.py]: Maybe you can to keep the print statement but only is the tutorials is not excluded

File: test/notebooks/test_notebooks.py [hunk 4]
```diff
@@ -15,76 +16,140 @@
...
+        if list_name in not_excluded_notebooks:
+            not_excluded_notebooks.remove(list_name)
+            res = pm.execute_notebook(notebook, "/dev/null", parameters={"epochs": 1}, timeout=300)
+            assert isinstance(res, nbformat.notebooknode.NotebookNode)
 
 
 def test_fl_with_trainconfig(isolated_filesystem, start_remote_server_worker_only, hook):
     os.chdir("advanced/Federated Learning with TrainConfig/")
+    notebook = "Introduction to TrainConfig.ipynb"
+    print(notebook)
```
Comment (LaRiffle) [inline test/notebooks/test_notebooks.py]: smae

**Comment by fdroessler** [comment]
Thanks, the print statements made it easier to debug but makes sense to exclude them in general.

**Comment by iamtrask** [comment]
Sadly a previous PR has created some merge conflicts. Can you attend?

**Pull request merged**